shared_model/interfaces/common_objects/types.hpp

```diff
@@ -92,6 +92,10 @@ namespace shared_model {
           boost::any_range<Transaction,
                            boost::random_access_traversal_tag,
                            const Transaction &>;
+      using TransactionForwardCollectionType =
```

Comment:
Please don't add types to this file. Any modification of this file triggers the rebuild of almost the whole project, because of transitive includes. This definition is better to be locally defined in transaction_sequence.hpp